### PR TITLE
Markdown: store if list is loose or not in the List

### PR DESCRIPTION
**Pull request by KristofferC**
Punt on updating the renders in base but do it for the Documenter HTML renderer: https://github.com/JuliaDocs/Documenter.jl/pull/685

### CODE REVIEW

**Comment by mortenpi** [comment]
Lists with multiple child blocks don't get the loose flag at the moment:

```julia
julia> Markdown.md"""
       - a
       - b

         a
       - c
       """.content
1-element Array{Any,1}:
 Markdown.List(Any[Any[Markdown.Paragraph(Any["a"])], Any[Markdown.Paragraph(Any["b"]), Markdown.Paragraph(Any["a"])], Any[Markdown.Paragraph(Any["c"])]], -1, false)
```

Is there anything specific holding back updating the print/show methods in Base to using the `.loose` flag? (I haven't looked at the code myself.)

Would you consider it a fix for #21922? Also, just for reference, #22071.

I think this is a good way to get the tight/loose lists in. Having `Paragraph`s wrapped around list items is actually consistent [with the AST from the CommonMark sample parser](http://spec.commonmark.org/dingus/?text=*%20A%0A*%20B%0A*%20C). It's then up to the renderer to display the paragraphs differently based on whether the list is loose or not (as you've done in Documenter).

**Comment by KristofferC** [comment]
Nothing preventing them from updating but there is a need to thread a parent through the every render call (like is done in Documenter) and getting Documenter correct seems like the most important thing. I will fix and add a test for the bug you mentioned.

**Comment by KristofferC** [comment]
Can someone explain how this new test passed everywhere but on Travis Mac. What is the platform difference?

File: stdlib/Markdown/src/Common/block.jl [hunk 1]
```diff
@@ -250,8 +250,9 @@ end
 mutable struct List
     items::Vector{Any}
     ordered::Int # `-1` is unordered, `>= 0` is ordered.
+    loose::Bool # TODO: Renderer's should use this field
 
-    List(x::AbstractVector, b::Integer) = new(x, b)
+    List(x::AbstractVector, b::Integer) = new(x, b, false)
```
Comment (KristofferC) [inline stdlib/Markdown/src/Common/block.jl]: Ooooooh, these two constructors below create `loose` with uninitialized memory. https://github.com/JuliaLang/julia/issues/24943 strikes again

**Comment by KristofferC** [comment]
Looks like the mac builder is trying to build gcc, presumably because some network error.

**Pull request merged**